compiler/src/iree/compiler/Dialect/Util/IR/UtilAttrs.cpp

```diff
@@ -227,7 +227,14 @@ static LogicalResult
 serializeResourceRawData(Location loc,
                          DenseResourceElementsAttr resourceElementsAttr,
                          llvm::raw_ostream &os) {
-  auto rawData = resourceElementsAttr.getRawHandle().getBlob()->getData();
+  auto *blob = resourceElementsAttr.getRawHandle().getBlob();
+  if (!blob) {
+    return mlir::emitError(loc)
+           << "resource data missing in input IR (removed by user?); "
+              "cannot serialize resource: "
+           << resourceElementsAttr << "\n";
+  }
+  auto rawData = blob->getData();
```

Comment:
Err... is this ASan error related?

https://github.com/iree-org/iree/actions/runs/10567109787/job/29275363530?pr=18361#step:4:11690
```
   892/1891 Test  #390: iree/compiler/ConstEval/test/jit_globals_vmvx_errors.mlir.test ....................................................***Failed    6.66 sec
  -- Testing: 1 tests, 1 workers --
  FAIL: IREE :: src/iree/compiler/ConstEval/test/jit_globals_vmvx_errors.mlir (1 of 1)
  ******************** TEST 'IREE :: src/iree/compiler/ConstEval/test/jit_globals_vmvx_errors.mlir' FAILED ********************
  Exit Code: 2
  
  Command Output (stderr):
  --
  RUN: at line 1: iree-opt --split-input-file --iree-consteval-jit-globals --iree-consteval-jit-target-device=vmvx --iree-consteval-jit-debug --verify-diagnostics /work/compiler/src/iree/compiler/ConstEval/test/jit_globals_vmvx_errors.mlir | FileCheck /work/compiler/src/iree/compiler/ConstEval/test/jit_globals_vmvx_errors.mlir
  + iree-opt --split-input-file --iree-consteval-jit-globals --iree-consteval-jit-target-device=vmvx --iree-consteval-jit-debug --verify-diagnostics /work/compiler/src/iree/compiler/ConstEval/test/jit_globals_vmvx_errors.mlir
  + FileCheck /work/compiler/src/iree/compiler/ConstEval/test/jit_globals_vmvx_errors.mlir
  ::: Rejected consteval initializer:
  util.initializer attributes {iree.compiler.consteval} {
    %cst = arith.constant dense<2.000000e+02> : tensor<5x6xf16>
    util.global.store %cst, @hoisted : tensor<5x6xf16>
    util.return
  }
  ::: Rejected consteval initializer:
  util.initializer attributes {iree.compiler.consteval} {
    %cst = arith.constant dense<2.000000e+02> : tensor<5x6xbf16>
    util.global.store %cst, @hoisted : tensor<5x6xbf16>
    util.return
  }
  ::: Rejected consteval initializer:
  util.initializer attributes {iree.compiler.consteval} {
    %cst = arith.constant dense<[2.000000e+02, 3.200000e+03]> : tensor<2xf64>
    util.global.store %cst, @hoisted : tensor<2xf64>
    util.return
  }
  ::: Rejected consteval initializer:
  util.initializer attributes {iree.compiler.consteval} {
    %cst = arith.constant dense<3> : tensor<5x6xi4>
    util.global.store %cst, @hoisted : tensor<5x6xi4>
    util.return
  }
  ::: COMPILING JIT (vmvx): module attributes {hal.device.targets = [#hal.device.target<"local", [#hal.executable.target<"vmvx", "vmvx-bytecode-fb", {ukernels = "none"}>]> : !hal.device], iree.consteval} {
    util.func public @jit_eval(%arg0: tensor<f32>) {
      util.return
    }
  }
  ::: Invoking jit_eval
  =================================================================
  ==59158==ERROR: AddressSanitizer: heap-use-after-free on address 0x51100044af00 at pc 0x7f693eaaf0de bp 0x7ffd8667b420 sp 0x7ffd8667b418
  WRITE of size 4 at 0x51100044af00 thread T0
      #0 0x7f693eaaf0dd in iree_vm_ref_object_release /work/runtime/src/iree/vm/ref.c:81:7
      #1 0x7f693e96a7fd in void iree::vm::ref_type_release<iree_hal_buffer_t>(iree_hal_buffer_t*) /work/runtime/src/iree/vm/ref_cc.h:48:3
      #2 0x7f693e96a7fd in iree::vm::ref<iree_hal_buffer_t>::~ref() /work/runtime/src/iree/vm/ref_cc.h:254:50
      #3 0x7f693e96a7fd in mlir::iree_compiler::ConstEval::FunctionCall::importSerializableAttr(mlir::Location, mlir::iree_compiler::IREE::Util::SerializableAttrInterface) /work/compiler/src/iree/compiler/ConstEval/Runtime.cpp:214:1
      #4 0x7f693e96b0b2 in mlir::iree_compiler::ConstEval::FunctionCall::addBufferViewArgumentAttr(mlir::Location, mlir::ShapedType, mlir::iree_compiler::IREE::Util::SerializableAttrInterface) /work/compiler/src/iree/compiler/ConstEval/Runtime.cpp:241:17
      #5 0x7f693e96b5dc in mlir::iree_compiler::ConstEval::FunctionCall::addArgument(mlir::Location, mlir::Attribute) /work/compiler/src/iree/compiler/ConstEval/Runtime.cpp
      #6 0x7f693b5cd6eb in mlir::iree_compiler::ConstEval::(anonymous namespace)::JitGlobalsPass::processFunctions(mlir::iree_compiler::ConstEval::CompiledBinary&, llvm::SmallVector<mlir::iree_compiler::ConstEval::(anonymous namespace)::JitFunctionDesc, 1u>&, mlir::ModuleOp, llvm::TimerGroup&) /work/compiler/src/iree/compiler/ConstEval/JitGlobals.cpp:739:27
      #7 0x7f693b5cd6eb in mlir::iree_compiler::ConstEval::(anonymous namespace)::JitGlobalsPass::runOnOperation() /work/compiler/src/iree/compiler/ConstEval/JitGlobals.cpp:859:16
      #8 0x7f693b6ea7b7 in mlir::detail::OpToOpPassAdaptor::run(mlir::Pass*, mlir::Operation*, mlir::AnalysisManager, bool, unsigned int)::$_1::operator()() const /work/third_party/llvm-project/mlir/lib/Pass/Pass.cpp:527:17
      #9 0x7f693b6ea7b7 in void llvm::function_ref<void ()>::callback_fn<mlir::detail::OpToOpPassAdaptor::run(mlir::Pass*, mlir::Operation*, mlir::AnalysisManager, bool, unsigned int)::$_1>(long) /work/third_party/llvm-project/llvm/include/llvm/ADT/STLFunctionalExtras.h:45:12
      #10 0x7f693b6ea7b7 in llvm::function_ref<void ()>::operator()() const /work/third_party/llvm-project/llvm/include/llvm/ADT/STLFunctionalExtras.h:68:12
      #11 0x7f693b6ea7b7 in void mlir::MLIRContext::executeAction<mlir::PassExecutionAction, mlir::Pass&>(llvm::function_ref<void ()>, llvm::ArrayRef<mlir::IRUnit>, mlir::Pass&) /work/third_party/llvm-project/mlir/include/mlir/IR/MLIRContext.h:275:7
      #12 0x7f693b6ea7b7 in mlir::detail::OpToOpPassAdaptor::run(mlir::Pass*, mlir::Operation*, mlir::AnalysisManager, bool, unsigned int) /work/third_party/llvm-project/mlir/lib/Pass/Pass.cpp:521:21
      #13 0x7f693b6ebd20 in mlir::detail::OpToOpPassAdaptor::runPipeline(mlir::OpPassManager&, mlir::Operation*, mlir::AnalysisManager, bool, unsigned int, mlir::PassInstrumentor*, mlir::PassInstrumentation::PipelineParentInfo const*) /work/third_party/llvm-project/mlir/lib/Pass/Pass.cpp:593:16
      #14 0x7f693b6f2cda in mlir::PassManager::runPasses(mlir::Operation*, mlir::AnalysisManager) /work/third_party/llvm-project/mlir/lib/Pass/Pass.cpp:904:10
      #15 0x7f693b6f2cda in mlir::PassManager::run(mlir::Operation*) /work/third_party/llvm-project/mlir/lib/Pass/Pass.cpp:884:60
      #16 0x7f693b6d470d in performActions(llvm::raw_ostream&, std::shared_ptr<llvm::SourceMgr> const&, mlir::MLIRContext*, mlir::MlirOptMainConfig const&) /work/third_party/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:413:17
      #17 0x7f693b6d3b90 in processBuffer(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, mlir::MlirOptMainConfig const&, mlir::DialectRegistry&, llvm::ThreadPoolInterface*) /work/third_party/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:486:9
      #18 0x7f693b6d3b90 in mlir::MlirOptMain(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, mlir::DialectRegistry&, mlir::MlirOptMainConfig const&)::$_0::operator()(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&) const /work/third_party/llvm-project/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp:561:12
      #19 0x7f693b6d3b90 in llvm::LogicalResult llvm::function_ref<llvm::LogicalResult (std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&)>::callback_fn<mlir::MlirOptMain(llvm::raw_ostream&, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, mlir::DialectRegistry&, mlir::MlirOptMainConfig const&)::$_0>(long, std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer>>, llvm::raw_ostream&) /work/third_party/llvm-project/llvm/include/llvm/ADT/STLFunctionalExtras.h:45:12
```

Reply:
lol must not be handling cleanup right - hooray asan!

Reply:
Broken in #17075.

Reply:
Oops :O